### PR TITLE
A bit more docs for matplotlib plotting

### PR DESCRIPTION
**Pull request by iosonofabio**
Addressing #398 with:

- a mention to matplotlib in the `ig.plot` docstring
- two mentions in the tutorial
- PEP8 linting of the tutorial, since we are there.

Let me know if we need anything else urgently (before next release) and I'll be happy to add it last minute!

### CODE REVIEW

File: doc/source/tutorial.rst [hunk 1]
```diff
@@ -753,6 +753,13 @@ nodes may be different on your machine since the layout is not deterministic):
 
    Our social network with the Kamada-Kawai layout algorithm
 
+If you prefer to use `matplotlib`_ as a default plotting engine, create an axes and use
+the ``target`` argument:
+
+>>> import matplotlib.pyplot as plt
+>>> fig, ax = plt.subplots()
+>>> plot(g, layout=layout, target=ax)
```
Comment (szhorvat) [inline doc/source/tutorial.rst:761]: It seems to me that this does not make `matplotlib` the _default_. It is necessary to explicitly specify the `target` argument, i.e. modify the default.

That brings up the question: what if I want `matplotlib` to be the default? Should this be possible? Does the current design even allow making it the default, give that `target` takes an already constructed axes object? I just want a plot to show up in my notebook, and I want it to be made with matplotlib.

Keep in mind that these comments are coming from a Python novice who has very minimal experience with matplotlib. Take it as such: the perspective of a novice (which I think is useful, but should not always be used as the basis of decisions).

File: src/igraph/drawing/__init__.py [hunk 2]
```diff
@@ -423,6 +423,10 @@ def plot(obj, target=None, bbox=(0, 0, 600, 600), *args, **kwds):
     @param target: the target where the object should be plotted. It can be one
       of the following types:
 
+        - C{matplotib.axes.Axes} -- a matplotlib/pyplot axes in which the
+          graph will be plotted. Drawing, saving to file, etc. is mostly
```
Comment (szhorvat) [inline src/igraph/drawing/__init__.py:427]: As a novice reading this, my immediate question is: what does "mostly" mean here?

File: doc/source/tutorial.rst (hunk 1 above)
Comment (szhorvat) [reply doc/source/tutorial.rst:761]: (I'm also hijacking this a bit to try to learn a bit more about Python ;-) )

**Comment by iosonofabio** [comment]
Fair enough, I improved the wording.

I's quite easy to let people swap the default, but requires a little care since it's a **default** change and could be set per session, config file, module-wide, and/or for each import of `igraph`.

We don't need it for this release, but we should keep it in mind if we want to push mpl for the next one.

**Comment by szhorvat** [comment]
What would you recommendation be if I wanted to have matplotlib as default today? Should I write a wrapper function for `plot` and let the default value of the `target` argument be an axes object created on the fly?

Once again, my comments here are more about my own learning than making suggestions.

**Comment by iosonofabio** [comment]
define your axes and use the function with that target. That's how I use matplotlib for everything, not just igraph. Nonetheless, most tutorials encourage you to let something called `pyplot` handle some black magic for you, but that's generally a bad idea inherited from matlab, alas...

File: doc/source/tutorial.rst (hunk 1 above)
Comment (ntamas) [reply doc/source/tutorial.rst]: Discussion continues below in the main comment section.

**Comment by ntamas** [comment]
@szhorvat So, I think that if you want Matplotlib to be the default, the best course of action is to override `plot()` with something like this:

```python
from igraph import plot as plot_original
import matplotlib.pyplot as plt

def plot(*args, **kwds):
    if "target" not in kwds:
        fig, ax = plt.subplots()
        kwds["target"] = ax
    return plot_original(*args, **kwds)
```

This basically injects a fresh Matplotlib axes object if you don't have a target.

**Pull request merged**